src/serial.rs

```diff
@@ -99,6 +99,33 @@ const DEFAULT_MODEM_CONTROL: u8 = MCR_OUT2_BIT;
 const DEFAULT_MODEM_STATUS: u8 = MSR_DSR_BIT | MSR_CTS_BIT | MSR_DCD_BIT;
 const DEFAULT_SCRATCH: u8 = 0x00;
 
+/// Defines a series of callbacks that are invoked in response to the occurrence of specific
```

Comment:
We should also describe a bit why we went with traits instead of concrete types.